edx_exams/apps/lti/views.py

```diff
@@ -97,12 +96,39 @@ def start_proctoring(request, attempt_id):
     return redirect(get_lti_1p3_launch_start_url(launch_data))
 
 
-def end_assessment(request, attempt_id):  # pragma: no cover
+@api_view(['GET'])
+@require_http_methods(['GET'])
+@authentication_classes((JwtAuthentication,))
+@permission_classes((IsAuthenticated,))
+def end_assessment(request, attempt_id):
     """
     LTI End Assessment
     """
-    # TODO: Here we'd do all the end of assessment things.
-    attempt = ExamAttempt.objects.get(pk=attempt_id)
+    # Verify that the attempt associated with attempt_id exists.
+    attempt = get_attempt_by_id(attempt_id)
+    if not attempt:
+        return Response(
+            status=status.HTTP_400_BAD_REQUEST,
+            data={'detail': f'Attempt with attempt_id={attempt_id} does not exist.'}
+        )
+
+    # Verify that that the requesting user is authorized to modify the attempt.
+    user = request.user
+    if attempt.user.id != user.id:
+        error_msg = (
+            f'user_id={attempt.user.id} attempted to update attempt_id={attempt.id} in '
+            f'course_id={attempt.exam.course_id} when starting an LTI launch but does not have access to it.'
+        )
+        error = {'detail': error_msg}
+        return Response(status=status.HTTP_403_FORBIDDEN, data=error)
+
+    update_attempt_status(attempt_id, ExamAttemptStatus.submitted)
+
+    # user is authenticated via JWT so use that to create a
+    # session with this service's authentication backend
+    request.user.backend = EDX_OAUTH_BACKEND
```

Comment:
I think we'd still need this code in the `end_assessment` view as well, based on https://github.com/edx/edx-exams/pull/59. It's probably unlikely their session will be invalidated or time out, but I figured there's no harm to log them in.

We definitely do not want to create a new session, but I couldn't find documentation about whether this would create a new session. Based on some testing, I don't think it should.

Reply:
This makes sense to me, but I'm not sure! Would probably still have someone take a look who's more familiar with this.

Reply:
Makes sense to me to include this since we can't 100% guarantee the user has a session anymore and the impact of it not existing is end_assessment errors out

Reply:
Great. Thanks, both!